ovh/resource_cloud_project_database_mongodb_user.go

```diff
@@ -0,0 +1,223 @@
+	config := meta.(*Config)
+	serviceName := d.Get("service_name").(string)
+	clusterId := d.Get("cluster_id").(string)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/mongodb/%s/user",
+		url.PathEscape(serviceName),
+		url.PathEscape(clusterId),
+	)
+	params := (&CloudProjectDatabaseMongodbUserCreateOpts{}).FromResource(d)
+	res := &CloudProjectDatabaseMongodbUserResponse{}
+
+	log.Printf("[DEBUG] Will create user: %+v for cluster %s from project %s", params, clusterId, serviceName)
+	err := config.OVHClient.Post(endpoint, params, res)
+	if err != nil {
+		return fmt.Errorf("calling Post %s with params %s:\n\t %q", endpoint, params, err)
+	}
+
+	log.Printf("[DEBUG] Waiting for database %s to be READY", clusterId)
+	err = waitForCloudProjectDatabaseReady(config.OVHClient, serviceName, "mongodb", clusterId, 60*time.Second, 10*time.Second)
+	if err != nil {
+		return fmt.Errorf("timeout while waiting database %s to be READY: %w", clusterId, err)
+	}
+	log.Printf("[DEBUG] database %s is READY", clusterId)
+
+	d.SetId(res.Id)
+	d.Set("password", res.Password)
+
+	return resourceCloudProjectDatabaseMongodbUserRead(d, meta)
+}
+
+func resourceCloudProjectDatabaseMongodbUserRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	serviceName := d.Get("service_name").(string)
+	clusterId := d.Get("cluster_id").(string)
+	id := d.Id()
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/mongodb/%s/user/%s",
+		url.PathEscape(serviceName),
+		url.PathEscape(clusterId),
+		url.PathEscape(id),
+	)
+	res := &CloudProjectDatabaseMongodbUserResponse{}
+
+	log.Printf("[DEBUG] Will read user %s from cluster %s from project %s", id, clusterId, serviceName)
+	if err := config.OVHClient.Get(endpoint, res); err != nil {
+		return helpers.CheckDeleted(d, err, endpoint)
+	}
+
+	for k, v := range res.ToMap() {
+		if k != "id" {
+			d.Set(k, v)
+		} else {
+			d.SetId(fmt.Sprint(v))
+		}
+	}
+
+	log.Printf("[DEBUG] Read user %+v", res)
+	return nil
+}
+
+func resourceCloudProjectDatabaseMongodbUserUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	serviceName := d.Get("service_name").(string)
+	clusterId := d.Get("cluster_id").(string)
+	id := d.Id()
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/mongodb/%s/user/%s",
+		url.PathEscape(serviceName),
+		url.PathEscape(clusterId),
+		url.PathEscape(id),
+	)
+	params := (&CloudProjectDatabaseMongodbUserUpdateOpts{}).FromResource(d)
+
+	log.Printf("[DEBUG] Will update user: %+v from cluster %s from project %s", params, clusterId, serviceName)
+	err := config.OVHClient.Put(endpoint, params, nil)
+	if err != nil {
+		return fmt.Errorf("calling Put %s with params %s:\n\t %q", endpoint, params, err)
+	}
+
+	log.Printf("[DEBUG] Waiting for database %s to be READY", clusterId)
+	err = waitForCloudProjectDatabaseReady(config.OVHClient, serviceName, "mongodb", clusterId, 60*time.Second, 10*time.Second)
```

Comment:
Here you should wait for the user to be properly created waiting for the status to be set at "READY" Same goes for other user engines